llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -3110,6 +3110,21 @@ KnownBits SelectionDAG::computeKnownBits(SDValue Op, const APInt &DemandedElts,
     }
     break;
   }
+  case ISD::STEP_VECTOR: {
+    const APInt &Step = Op.getConstantOperandAPInt(0);
+
+    if (Step.isPowerOf2())
+      Known.Zero.setLowBits(Step.logBase2());
+
+    const Function &F = getMachineFunction().getFunction();
+    const APInt MaxNumElts = getVScaleRange(&F, BitWidth).getUnsignedMax() *
```

Comment:
The element count multiply can also overflow.

Reply:
I think the min number of elements itself can also overflow, so I've added a check that it fits into the bitwidth